### ISSUE TITLE
Installed targets lack an include interface for pdal headers

### ISSUE THREAD

**Issue by mathstuf** [PDAL/PDAL#2262]
The installed targets from the CMake build lack `INTERFACE_INCLUDE_DIRECTORIES` which means that `target_include_directories` is required in addition to `target_link_libraries`. Something like [this](https://github.com/libLAS/libLAS/pull/153/files#diff-95e351a3805a1dafa85bf20b81d086e6R236) should be done. I'd do it, but PDAL has a much more complex CMake setup that I'm not sure exactly where best to put it.

**Comment by abellgithub** [comment]
It would be best if you could report a specific problem that you're having so that we can fix it.  Knowing how your suggested change might impact others is difficult to know with what you've submitted.

**Comment by mathstuf** [comment]
PDAL offers imported targets when `find_package(PDAL)` is done (via `install(EXPORT)`). These targets should be usable as-is without having to also use `PDAL_INCLUDE_DIRS` by doing, somewhere:

```cmake
target_include_directories(pdal_target
    INTERFACE
        $<INSTALL_INTERFACE:include>)
```

**Comment by abellgithub** [comment]
Fixed with #2981
